telemetry/definitions/commonDefinitions.json

```diff
@@ -2602,62 +2611,53 @@
             "metadata": [{ "type": "amazonqConversationId" }]
         },
         {
-            "name": "amazonq_approachInvoke",
-            "description": "Emitted when the user started Approach process and captures if the proccess succeeded or failed.",
+            "name": "amazonq_endChat",
+            "description": "Captures end of the conversation with amazonq /dev",
             "metadata": [
                 { "type": "amazonqConversationId" },
-                { "type": "result" },
-                { "type": "reason", "required": false }
+                { "type": "amazonqEndOfTheConversationLatency", "required": false }
             ]
         },
         {
-            "name": "amazonq_approachIteration",
-            "description": "User recieved a result in Approach Iteration stage",
+            "name": "amazonq_approachInvoke",
+            "description": "Captures Approach generation process",
             "metadata": [
                 { "type": "amazonqConversationId" },
                 { "type": "amazonqGenerateApproachIteration" },
-                { "type": "amazonqGenerateApproachLatency" }
+                { "type": "amazonqGenerateApproachLatency" },
+                { "type": "result" },
```

Comment:
Do we need to explicitly add these? I thought they came from the metric base

Reply:
let's keep for readability purposes

Reply:
All metrics have result/reason (see `MetricBase` interface) so you will get autocompletion for result/reason. I think the goal is to keep these minimal so we don't need to add `result` since it is implicit, I will look to update docs somewhere to indicate this